### PR TITLE
Optimize netloc extraction

### PR DESCRIPTION
**Pull request by elliotwutingfeng**
**SCHEME_RE** can be replaced with an **if-else** equivalent for tangible speed improvement.

Benchmarks of this optimization together with improvements in #283

Python 3.10, Linux x64, Ryzen 7 5800X

```python
import tldextract

%timeit tldextract.extract("")
%timeit tldextract.extract("com")
%timeit tldextract.extract("example\u3002com")
%timeit tldextract.extract("subdomain\uff0eexample\uff61com")
%timeit tldextract.extract("a\u3002very\uff0elong\uff61subdomain\u3002example\uff0ecom")
%timeit tldextract.extract("an\uff61even\u3002longer\uff0eand\uff61complex\u3002subdomain\uff0eexample\uff61com")
%timeit tldextract.extract("https://a\u3002b\uff0ec\uff61d\u3002e\uff0ef\uff61g\u3002h\uff0ei\uff61j\u3002k\uff0el\uff61m\u3002n\uff0eoo\uff61pp\u3002qqq\uff0errrr\uff61ssssss\u3002tttttttt\uff0euuuuuuuuuuu\uff61vvvvvvvvvvvvvvv\u3002wwwwwwwwwwwwwwwwwwwwww\uff0exxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\uff61yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy\u3002zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz.tw")
%timeit tldextract.extract("\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\u3002\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff0e\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61\uff61")
```

```python
1.72 µs ± 14.4 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
1.75 µs ± 15.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
2.34 µs ± 9.41 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
2.41 µs ± 4.97 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
2.6 µs ± 4.92 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
2.67 µs ± 18.9 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
3.76 µs ± 13.4 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
4.49 µs ± 21.2 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

## Changes

- Replaced **SCHEME_RE** with faster **if-else** equivalent

### CODE REVIEW

File: tldextract/remote.py [hunk 1]
```diff
@@ -29,6 +25,23 @@ def lenient_netloc(url: str) -> str:
...
+def lenient_netloc(url: str) -> str:
+    """Extract the netloc of a URL-like string, similar to the netloc attribute
+    returned by urllib.parse.{urlparse,urlsplit}, but extract more leniently,
+    without raising errors."""
+
+    double_slashes_start = url.find("//")
+    if double_slashes_start == 0:
+        return extract_netloc(url[2:])
+    if (
+        double_slashes_start < 2
```
Comment (john-kurkowski) [inline tldextract/remote.py:36]: What conditions are these 3 lines testing? If I delete either or both of the 2nd and 3rd lines, tests still pass. Can you add tests?

**Review by john-kurkowski — COMMENTED** [review]
Nice! 

I took your code in a little different direction, 1 public function in the module calling 1 private utility function, vs. 2 public functions with 1 wrapping the other.

File: tldextract/remote.py (hunk 1 above)
Comment (elliotwutingfeng) [reply tldextract/remote.py:36]:
```python
>       assert_extract(
            "//mail.google.com/mail", ("mail.google.com", "mail", "google", "com")
        )
```

The above test fails on Python 3.10 after commenting out these 2 lines. Are you getting the same results?

```python
    # if double_slashes_start == 0:
    #     return url[2:]
```

Comment (elliotwutingfeng) [reply tldextract/remote.py:36]: I think you meant

```python
        # or not url[double_slashes_start - 1] == ":"
        # or set(url[: double_slashes_start - 1]) - scheme_chars_set
```

Comment (elliotwutingfeng) [reply tldextract/remote.py:36]: I've added 2 tests for the above 2 conditions.

`tldextract.extract("a#//example.com")` output is the same as the regex version.

**Comment by john-kurkowski** [comment]
Thank you!

**Pull request merged**